src/extensions/askUserForParam.ts

```diff
@@ -116,6 +129,81 @@ export async function askForParam(paramSpec: Param): Promise<string> {
   return response;
 }
 
+async function promptSecret(
+  projectId: string,
+  instanceId: string,
+  paramSpec: Param
+): Promise<string> {
+  if (paramSpec.reconfiguring) {
+    const action = await promptOnce({
+      type: "list",
+      message: `Choose what you would like to do with this secret:`,
+      choices: [
+        { name: "Leave unchanged", value: SecretUpdateAction.LEAVE },
+        { name: "Set new value", value: SecretUpdateAction.SET_NEW },
+      ],
+    });
+    switch (action) {
+      case SecretUpdateAction.SET_NEW:
+        let secret;
+        let secretName;
+        if (paramSpec.default) {
+          secret = secretManagerApi.parseSecretResourceName(paramSpec.default);
+          secretName = secret.name;
+        } else {
+          secretName = await generateSecretName(projectId, instanceId, paramSpec.param);
+        }
+
+        const secretValue = await promptOnce({
+          name: paramSpec.param,
+          type: "input",
+          message: `Enter new value for ${paramSpec.label.trim()}. This secret will be stored in Cloud Secret Manager as ${secretName}:`,
+        });
+        if (!secret) {
+          secret = await secretManagerApi.createSecret(projectId, secretName);
```

Comment:
As is, I think you will never hit this 'create a new secret' path - you set paramSpec.reconfiguring to true when you set paramSpec.default, so !secret will never be true.

Reply:
This is a case when old default value is empty.